src/cpp/server/backend_metric_recorder.cc

```diff
@@ -41,16 +42,23 @@ grpc_core::TraceFlag grpc_backend_metric_trace(false, "backend_metric");
 namespace grpc {
 namespace experimental {
 
+std::unique_ptr<ServerMetricRecorder> ServerMetricRecorder::Create() {
+  return std::unique_ptr<ServerMetricRecorder>(new ServerMetricRecorder());
+}
+
 ServerMetricRecorder::ServerMetricRecorder() {
   // Starts with an empty result.
-  metric_state_ = std::make_shared<const BackendMetricDataState>();
+  metric_state_ =
+      std::make_shared<const ServerMetricRecorder::BackendMetricDataState>();
 }
 
 void ServerMetricRecorder::UpdateBackendMetricDataState(
     std::function<void(BackendMetricData*)> updater) {
   internal::MutexLock lock(&mu_);
-  auto new_state = std::make_shared<BackendMetricDataState>(*metric_state_);
-  updater(&(new_state->data));
+  auto new_state =
+      std::make_shared<ServerMetricRecorder::BackendMetricDataState>(
```

Comment:
No need to say `ServerMetricRecorder::` here, since you're already in that class.

Reply:
Done.